src/sqlfluff/rules/L031.py

```diff
@@ -71,12 +76,33 @@ class Rule_L031(BaseRule):
 
     """
 
+    config_keywords = ["force_enable"]
+    _dialects_disabled_by_default = ["bigquery"]
+
     def _eval(self, context: RuleContext) -> Optional[LintResult]:
         """Identify aliases in from clause and join conditions.
 
         Find base table, table expressions in join, and other expressions in select
         clause and decide if it's needed to report them.
         """
+        # Issue 2810: BigQuery has some tricky expectations (apparently not
+        # documented, but subject to change, e.g.:
+        # https://www.reddit.com/r/bigquery/comments/fgk31y/new_in_bigquery_no_more_backticks_around_table/)
+        # about whether backticks are required (and whether the query is valid
+        # or not, even with them), depending on whether the GCP project name is
+        # present, or just the dataset name. Since SQLFluff doesn't have access
+        # to BigQuery when it is looking at the query, it would be complex for
+        # this rule to do the right thing. For now, the rule simply disables
+        # itself.
+        # Config type hints
+        self.force_enable: bool
+
```

Comment:
```suggestion
        # Config type hints
        self.force_enable: bool

        # Issue 2810: BigQuery has some tricky expectations (apparently not
        # documented, but subject to change, e.g.:
        # https://www.reddit.com/r/bigquery/comments/fgk31y/new_in_bigquery_no_more_backticks_around_table/)
        # about whether backticks are required (and whether the query is valid
        # or not, even with them), depending on whether the GCP project name is
        # present, or just the dataset name. Since SQLFluff doesn't have access
        # to BigQuery when it is looking at the query, it would be complex for
        # this rule to do the right thing. For now, the rule simply disables
        # itself.
```